lib/core/src/lib/datatable/components/datatable-cell/datatable-cell.component.ts

```diff
@@ -108,6 +93,24 @@ export class DataTableCellComponent implements OnInit, OnDestroy {
         }
     }
 
+    protected subscribeToRowUpdates() {
+        if (!this.dataTableService) {
+            return;
+        }
+        this.dataTableService.rowUpdate.pipe(takeUntil(this.onDestroy$)).subscribe((data) => {
+            if (data?.id && this.row.id === data.id) {
+                if (this.row.obj && data.obj) {
+                    this.row.obj = data.obj;
+                    this.row['cache'][this.column.key] = this.column.key
+                        .split('.')
+                        .reduce((source, key) => (source ? source[key] : ''), data.obj);
+
+                    this.updateValue();
+                }
+            }
+        });
+    }
+
```

Comment:
i think we can split it more to make it more understandable, something like:

```suggestion
private subscribeToRowUpdates(): void {
    if (!this.dataTableService || !this.row.obj) {
        return;
    }

    this.dataTableService.rowUpdate.pipe(takeUntil(this.onDestroy$)).subscribe((data) => {
        if (data?.id === this.row?.id && data.obj) {
            this.row.obj = data.obj;
            this.row['cache'][this.column.key] = this.getNestedPropertyValue(data.obj, this.column.key);
            
            this.updateValue();
            }
        }
    });
}

private getNestedPropertyValue(obj: any, path: string) {
    return path.split('.').reduce((source, key) => (source ? source[key] : ''), obj);
}
```

Reply:
Good point, I'll change it, thx